### PR TITLE
Enhance - `azurerm_api_management_api` - support `contact`, `license`

### PR DESCRIPTION
**Pull request by xuzhang3**
Support config `contact`, `license`, `terms_of_service_url`

```log
=== RUN   TestAccApiManagementApi_basic
=== PAUSE TestAccApiManagementApi_basic
=== RUN   TestAccApiManagementApi_wordRevision
=== PAUSE TestAccApiManagementApi_wordRevision
=== RUN   TestAccApiManagementApi_blankPath
=== PAUSE TestAccApiManagementApi_blankPath
=== RUN   TestAccApiManagementApi_version
=== PAUSE TestAccApiManagementApi_version
=== RUN   TestAccApiManagementApi_oauth2Authorization
=== PAUSE TestAccApiManagementApi_oauth2Authorization
=== RUN   TestAccApiManagementApi_openidAuthentication
=== PAUSE TestAccApiManagementApi_openidAuthentication
=== RUN   TestAccApiManagementApi_requiresImport
=== PAUSE TestAccApiManagementApi_requiresImport
=== RUN   TestAccApiManagementApi_graphql
=== PAUSE TestAccApiManagementApi_graphql
=== RUN   TestAccApiManagementApi_soap
=== PAUSE TestAccApiManagementApi_soap
=== RUN   TestAccApiManagementApi_websocket
=== PAUSE TestAccApiManagementApi_websocket
=== RUN   TestAccApiManagementApi_soapPassthrough
=== PAUSE TestAccApiManagementApi_soapPassthrough
=== RUN   TestAccApiManagementApi_subscriptionRequired
=== PAUSE TestAccApiManagementApi_subscriptionRequired
=== RUN   TestAccApiManagementApi_importSwagger
=== PAUSE TestAccApiManagementApi_importSwagger
=== RUN   TestAccApiManagementApi_importWsdl
=== PAUSE TestAccApiManagementApi_importWsdl
=== RUN   TestAccApiManagementApi_importUpdate
=== PAUSE TestAccApiManagementApi_importUpdate
=== RUN   TestAccApiManagementApi_complete
=== PAUSE TestAccApiManagementApi_complete
=== RUN   TestAccApiManagementApi_cloneApi
=== PAUSE TestAccApiManagementApi_cloneApi
=== RUN   TestAccApiManagementApi_createNewVersionFromExisting
=== PAUSE TestAccApiManagementApi_createNewVersionFromExisting
=== RUN   TestAccApiManagementApi_createRevisionFromExisting
=== PAUSE TestAccApiManagementApi_createRevisionFromExisting
=== RUN   TestAccApiManagementApi_createRevisionFromExistingRevision
=== PAUSE TestAccApiManagementApi_createRevisionFromExistingRevision
=== RUN   TestAccApiManagementApi_contact
=== PAUSE TestAccApiManagementApi_contact
=== CONT  TestAccApiManagementApi_basic
=== CONT  TestAccApiManagementApi_subscriptionRequired
=== CONT  TestAccApiManagementApi_cloneApi
=== CONT  TestAccApiManagementApi_importUpdate
=== CONT  TestAccApiManagementApi_complete
=== CONT  TestAccApiManagementApi_importWsdl
=== CONT  TestAccApiManagementApi_importSwagger
=== CONT  TestAccApiManagementApi_createRevisionFromExisting
--- PASS: TestAccApiManagementApi_basic (542.01s)
=== CONT  TestAccApiManagementApi_createNewVersionFromExisting
--- PASS: TestAccApiManagementApi_cloneApi (549.29s)
=== CONT  TestAccApiManagementApi_contact
--- PASS: TestAccApiManagementApi_importWsdl (549.36s)
=== CONT  TestAccApiManagementApi_requiresImport
--- PASS: TestAccApiManagementApi_importSwagger (550.12s)
=== CONT  TestAccApiManagementApi_soapPassthrough
--- PASS: TestAccApiManagementApi_subscriptionRequired (550.60s)
=== CONT  TestAccApiManagementApi_websocket
--- PASS: TestAccApiManagementApi_createRevisionFromExisting (551.19s)
=== CONT  TestAccApiManagementApi_soap
--- PASS: TestAccApiManagementApi_complete (564.15s)
=== CONT  TestAccApiManagementApi_graphql
--- PASS: TestAccApiManagementApi_importUpdate (678.92s)
=== CONT  TestAccApiManagementApi_version
--- PASS: TestAccApiManagementApi_contact (518.18s)
=== CONT  TestAccApiManagementApi_openidAuthentication
--- PASS: TestAccApiManagementApi_soapPassthrough (520.43s)
=== CONT  TestAccApiManagementApi_oauth2Authorization
--- PASS: TestAccApiManagementApi_soap (525.18s)
=== CONT  TestAccApiManagementApi_blankPath
--- PASS: TestAccApiManagementApi_createNewVersionFromExisting (540.40s)
=== CONT  TestAccApiManagementApi_wordRevision
--- PASS: TestAccApiManagementApi_graphql (523.62s)
=== CONT  TestAccApiManagementApi_createRevisionFromExistingRevision
--- PASS: TestAccApiManagementApi_requiresImport (564.24s)
--- PASS: TestAccApiManagementApi_version (537.72s)
--- PASS: TestAccApiManagementApi_blankPath (435.38s)
--- PASS: TestAccApiManagementApi_wordRevision (502.94s)
--- PASS: TestAccApiManagementApi_oauth2Authorization (520.85s)
--- PASS: TestAccApiManagementApi_openidAuthentication (531.33s)
--- PASS: TestAccApiManagementApi_createRevisionFromExistingRevision (534.59s)
--- PASS: TestAccApiManagementApi_websocket (5274.84s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement 5876.199s
```

### CODE REVIEW

File: internal/services/apimanagement/api_management_api_resource.go [hunk 1]
```diff
@@ -100,6 +100,32 @@ func resourceApiManagementApi() *pluginsdk.Resource {
...
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"email": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
```
Comment (katbyte) [inline internal/services/apimanagement/api_management_api_resource.go:113]: can we validate this as an email

**Review by katbyte — CHANGES_REQUESTED** [review]
have some test failures:
```
------- Stdout: -------
=== RUN   TestAccApiManagementApiTagDescription_basic
=== PAUSE TestAccApiManagementApiTagDescription_basic
=== CONT  TestAccApiManagementApiTagDescription_basic
    testing_new.go:84: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: deleting Api Tag Descriptions: (Tag Description Name "acctest-Tag-220922142415133918" / Api Name "acctestapi-220922142415133918" / Service Name "acctestAM-220922142415133918" / Resource Group "acctestRG-220922142415133918"): apimanagement.APITagDescriptionClient#Delete: Failure responding to request: StatusCode=412 -- Original Error: autorest/azure: Service returned an error. Status=412 Code="PreconditionFailed" Message="Exception of type 'Microsoft.WindowsAzure.ApiManagement.Management.Core.Exceptions.PreconditionFailedException' was thrown."
        
--- FAIL: TestAccApiManagementApiTagDescription_basic (216.48s)
FAIL
```

**Comment by xuzhang3** [comment]
> have some test failures:
> 
> ```
> ------- Stdout: -------
> === RUN   TestAccApiManagementApiTagDescription_basic
> === PAUSE TestAccApiManagementApiTagDescription_basic
> === CONT  TestAccApiManagementApiTagDescription_basic
>     testing_new.go:84: Error running post-test destroy, there may be dangling resources: exit status 1
>         
>         Error: deleting Api Tag Descriptions: (Tag Description Name "acctest-Tag-220922142415133918" / Api Name "acctestapi-220922142415133918" / Service Name "acctestAM-220922142415133918" / Resource Group "acctestRG-220922142415133918"): apimanagement.APITagDescriptionClient#Delete: Failure responding to request: StatusCode=412 -- Original Error: autorest/azure: Service returned an error. Status=412 Code="PreconditionFailed" Message="Exception of type 'Microsoft.WindowsAzure.ApiManagement.Management.Core.Exceptions.PreconditionFailedException' was thrown."
>         
> --- FAIL: TestAccApiManagementApiTagDescription_basic (216.48s)
> FAIL
> ```

Cannot reproduce this error.
```log
=== RUN   TestAccApiManagementApiTagDescription_basic
=== PAUSE TestAccApiManagementApiTagDescription_basic
=== CONT  TestAccApiManagementApiTagDescription_basic
--- PASS: TestAccApiManagementApiTagDescription_basic (689.95s)
PASS
```

**Comment by katbyte** [comment]
test failures still:
image

**Comment by xuzhang3** [comment]
@katbyte   rerun the accTest a few times, all passed.
image

**Review by katbyte — APPROVED** [review]
thanks, LGTM 🦺

**Pull request merged**